File: web-sources/S8-pkgs-ui-carbide/forms/light/LightFormWrapper.js


import { S8 } from '/S8-api/S8Context.js';
import { LightFormElement } from './LightFormElement.js';
import { S8Object } from '/S8-api/S8Object.js';


S8.page.CSS_import("/S8-pkgs-ui-carbide/forms/light/LightForm.css");

/**
 * 
 */
export class LightFormWrapper extends S8Object {


    /**
     * @type{LightFormElement}
     */
    root;


    /**
     * 
     */
    constructor(){
        super();

        this.node = document.createElement("div");
        this.node.classList.add("lightform-box");

        this.node.addEventListener("scroll", function(event){ event.stopPropagation(); });
    }


    S8_render(){ /* continuous rendering approach... */ }

    /**
     * 
     * @param {*} root 
     */
    S8_set_root(root) {
        this.root = root;
        this.node.appendChild(root.getEnvelope());
    }


    getEnvelope() {
        return this.node;
    }

    /** @returns{S8Object[]} */
    getComponents() {
        return [this.root];
    }

    S8_dispose(){ /* TODO */ }
}

File: sources/com/s8/pkgs/ui/carbide/forms/light/LightFormWrapper.java
package com.s8.pkgs.ui.carbide.forms.light;

import com.s8.api.web.S8WebFront;
import com.s8.pkgs.ui.carbide.WebSources;
import com.s8.pkgs.ui.carbide.HTML_S8WebNode;


/**
 * 
 * @author pierreconvert
 *
 */
public class LightFormWrapper extends HTML_S8WebNode {

	
	public final static String WEBPATH = WebSources.ROOT_WEBPATH + "/forms/light";
	
	
	/**
	 * 
	 */
	public LightFormWrapper(S8WebFront front) {
		super(front, WEBPATH + "/LightFormWrapper");
	}
	
	
	/**
	 * 
	 * @param element
	 */
	public void setRoot(LightFormElement element) {
		vertex.outbound().setObjectField("root", element);
	}
	
	
	
}
